lib/jekyll/static_file.rb

```diff
@@ -97,6 +97,7 @@ def write(dest)
 
     def to_liquid
       {
+        "name"          => File.basename(name, extname),
```

Comment:
From what I understand, in the initialize function, name is the complete name of the file including the extension.

I just modified the output in Liquid, to be able to get only the name without the extension, already available as extname. I did not want to modify the current behavior, as it would break all current tests. 

As a total beginner, I have no clue how to properly achieve this. I just thought that we got to keep `name` as it is, and just add `filename` in the `to_liquid` function.